### PR TITLE
Global error handler cleanup - Counter and Observable counter creation

### PR DESCRIPTION
**Pull request by lalitb**
Fixes #
Design discussion issue (if applicable) #

## Changes

Please provide a brief description of the changes here.

## Merge requirement checklist

* [ ] [CONTRIBUTING](https://github.com/open-telemetry/opentelemetry-rust/blob/main/CONTRIBUTING.md) guidelines followed
* [ ] Unit tests added/updated (if applicable)
* [ ] Appropriate `CHANGELOG.md` files updated for non-trivial, user-facing changes
* [ ] Changes in public API reviewed (if applicable)

### CODE REVIEW

File: opentelemetry-sdk/src/metrics/meter.rs [hunk 1]
```diff
@@ -75,14 +76,20 @@ impl SdkMeter {
...
         let validation_result = validate_instrument_config(builder.name.as_ref(), &builder.unit);
         if let Err(err) = validation_result {
-            global::handle_error(err);
+            otel_error!(
+                name: "InstrumentCreationFailed",
+                meter_name = self.scope.name.as_ref(),
+                instrument_name = builder.name.as_ref(),
+                message = "Measurements from this counter will be ignored.",
+                reason = format!("{}", err)
+            );
```
Comment (lalitb) [inline opentelemetry-sdk/src/metrics/meter.rs:85]: Cijo's comment:
otel_error!(
name: "InstrumentCreationFailed",
meter_name = self.scope.name.as_ref(),
instrument_name = builder.name.as_ref(),
message = "Measurements from this instrument will be ignored."
reason = fmt(err))

^ I prefer this version. Please see if this is better for end users.

Comment (lalitb) [reply opentelemetry-sdk/src/metrics/meter.rs:85]: Utkarsh comment: 
change
`"Measurements from this instrument will be ignored."` to
`""Measurements from this counter will be ignored.",`

Comment (cijothomas) [inline opentelemetry-sdk/src/metrics/meter.rs:85]: 2024-10-24T05:32:23.524520Z ERROR main opentelemetry_sdk meter_name="mylibraryname" instrument_name="-my_counter" Measurements from this counter will be ignored. reason="Invalid instrument configuration: instrument name must start with an alphabetic character"

^ This is what we get when I used fmt module. The reason is a slightly extra verbose:
Current: _Invalid instrument configuration: instrument name must start with an alphabetic character_
Proposal: _Instrument name must start with an alphabetic character_

This is a minor thing we can change in the future.

File: opentelemetry-sdk/src/metrics/meter.rs [hunk 2]
```diff
@@ -107,19 +120,30 @@ impl SdkMeter {
     {
         let validation_result = validate_instrument_config(builder.name.as_ref(), &builder.unit);
         if let Err(err) = validation_result {
-            global::handle_error(err);
+            otel_error!(
+                name: "InstrumentCreationFailed", 
+                meter_name = self.scope.name.as_ref(),
+                instrument_name = builder.name.as_ref(),
+                message = "Measurements from this observable counter will be ignored.",
```
Comment (cijothomas) [inline opentelemetry-sdk/src/metrics/meter.rs:127]: nit: if we are doing different message for each kind of instrument, then for observable, we need to call out that the registered callbacks will not be invoked.

**Review by cijothomas — APPROVED** [review]
Left a nit suggestion, but LGTM for merging.

File: opentelemetry-sdk/src/metrics/meter.rs (hunk 2 above)
Comment (lalitb) [reply opentelemetry-sdk/src/metrics/meter.rs]: This is good point. Have updated it.

**Review by utpilla — APPROVED** [review]

**Review by TommyCpp — APPROVED** [review]

**Pull request merged**